tests/it/main.rs

```diff
@@ -475,7 +466,7 @@ fn string_escapes() {
 
 #[test]
 fn nonexistent_current_directory() {
-    let sh = setup();
+    let mut sh = setup();
     sh.change_dir("nonexistent");
     let err = cmd!(sh, "ls").run().unwrap_err();
     let message = err.to_string();
```

Comment:
This is not a small ask, but consider whether, to better judge if the APIs we add are ergonomic, we want to add an extended example.

I imagine something like `examples/api_walkthrough` which show cases all common methods of the API, using a somewhat real-world-ish examples. I imagine the example can be organized as a set of fucntion, where each function does some "real world" task, and the set of functions together covers most of the public APIs. 

The benefits here wouold be:

* for us, we'd get a better feel for the API looking at the call-site, rather than the definition site
* for users, they'll get another excellent documentation option.